### ISSUE TITLE
Add an exhaustivity checker

### ISSUE THREAD

**Issue by vendethiel** [vendethiel/ADT.ex#3]

**Comment by vendethiel** [comment]
Exposed module's variants via 050efdb5c3421c3f9a66f2b60fc29a7a35d047cf.

**Comment by vendethiel** [comment]
```
> quote do case foo do 1 when %Foo{} -> 2 end end   
{:case, [],
 [{:foo, [], Elixir},
  [do: [{:->, [],
     [[{:%, [], [{:__aliases__, [alias: false], [:Foo]}, {:%{}, [], []}]}],
      2]}]]]}
> quote do case foo do %NS.Foo{} -> 2 end end
{:case, [],
 [{:foo, [], Elixir},
  [do: [{:->, [],
     [[{:%, [], [{:__aliases__, [alias: false], [:NS, :Foo]}, {:%{}, [], []}]}],
      2]}]]]}
```

**Comment by vendethiel** [comment]
fixed by #6
